samples/04.0-file-transfer/integration-tests/src/testFixtures/java/org/eclipse/dataspaceconnector/extension/sample/test/FileTransferSampleTestCommon.java

```diff
@@ -44,11 +44,11 @@ public class FileTransferSampleTestCommon {
     //region constant test settings
     static final String INITIATE_CONTRACT_NEGOTIATION_URI = "http://localhost:9192/api/v1/data/contractnegotiations";
     static final String LOOK_UP_CONTRACT_AGREEMENT_URI = "http://localhost:9192/api/v1/data/contractnegotiations/{id}";
-    static final String INITIATE_TRANSFER_PROCESS_URI = "http://localhost:9192/api/v1/data/transferprocess";
+    public static final String TRANSFER_PROCESS_URI = "http://localhost:9192/api/v1/data/transferprocess";
     static final String CONTRACT_OFFER_FILE_PATH = "samples/04.0-file-transfer/contractoffer.json";
     static final String TRANSFER_FILE_PATH = "samples/04.0-file-transfer/filetransfer.json";
-    static final String API_KEY_HEADER_KEY = "X-Api-Key";
-    static final String API_KEY_HEADER_VALUE = "password";
+    public static final String API_KEY_HEADER_KEY = "X-Api-Key";
+    public static final String API_KEY_HEADER_VALUE = "password";
```

Comment:
Instead of exposing these settings you could experiment with splitting the methods in `FileTransferSampleTestCommon` more fine granularly so that they can be reused better. For instance:

```
ValidatableResponse initiateContractNegotiation() {
        return RestAssured
                .given()
                .headers(API_KEY_HEADER_KEY, API_KEY_HEADER_VALUE)
                .contentType(ContentType.JSON)
                .body(new File(TestUtils.findBuildRoot(), CONTRACT_OFFER_FILE_PATH))
            .when()
                .post(INITIATE_CONTRACT_NEGOTIATION_URI)
            .then();
    }
```

This could be reused by all samples to initiate a contract negotiation, and then define expectations on top of it.

Reply:
Here you have some sample code for what I am suggesting: https://github.com/agera-edc/DataSpaceConnector/commit/258d5c2870e836c2a773fadcce74d6decce52937